packages/flutter/lib/src/material/toggle_buttons.dart

```diff
@@ -0,0 +1,1121 @@
+/// that this assumes that [isSelected] was properly initialized with one
+/// selection.
+/// ```dart
+/// ToggleButtons(
+///   children: <Widget>[
+///     Icon(Icons.ac_unit),
+///     Icon(Icons.call),
+///     Icon(Icons.cake),
+///   ],
+///   onPressed: (int index) {
+///     int count = 0;
+///     isSelected.forEach((bool val) {
+///       if (val) count++;
+///     });
+///
+///     if (isSelected[index] && count < 2)
+///       return;
+///
+///     setState(() {
+///       isSelected[index] = !isSelected[index];
+///     });
+///   },
+///   isSelected: isSelected,
+/// ),
+/// ```
+///
+/// ## ToggleButton Borders
+/// The toggle buttons, by default, have a solid, 1 dp border surrounding itself
+/// and separating each button. The toggle button borders' color, width, and
+/// corner radii are configurable.
+///
+/// The [selectedBorderColor] determines the border's color when the button is
+/// selected, while [disabledBorderColor] determines the border's color when
+/// the button is disabled. [borderColor] is used when the button is enabled.
+///
+/// To remove the border, set [borderWidth] to null. Setting [borderWidth] to
+/// 0.0 results in a hairline border. For more information on hairline borders,
+/// see [BorderSide.width].
+///
+/// See also:
+///
+///  * <https://material.io/design/components/buttons.html#toggle-button>
+class ToggleButtons extends StatelessWidget {
+  /// Creates a horizontal set of toggle buttons.
+  ///
+  /// It displays its widgets provided in a [List] of [children] horizontally.
+  /// The state of each button is controlled by [isSelected], which is a list
+  /// of bools that determine if a button is in an active, disabled, or
+  /// selected state. They are both correlated by their index in the list.
+  /// The length of [isSelected] has to match the length of the [children]
+  /// list.
+  ///
+  /// Both [children] and [isSelected] properties arguments are required.
+  ///
+  /// [isSelected] values must be non-null. [focusNodes] must be null or a
+  /// list of non-null nodes. [renderBorder] must not be null.
+  const ToggleButtons({
+    Key key,
+    @required this.children,
+    @required this.isSelected,
+    this.onPressed,
+    this.color,
+    this.selectedColor,
+    this.disabledColor,
+    this.fillColor,
+    this.focusColor,
+    this.highlightColor,
+    this.hoverColor,
+    this.splashColor,
+    this.focusNodes,
+    this.renderBorder = true,
+    this.borderColor,
+    this.selectedBorderColor,
+    this.disabledBorderColor,
+    this.borderRadius,
+    this.borderWidth,
+  }) :
+    assert(children != null),
+    assert(isSelected != null),
+    assert(children.length == isSelected.length),
+    super(key: key);
+
+  static const double _defaultBorderWidth = 1.0;
+
+  /// The corresponding widget values in the toggle buttons.
+  ///
+  /// These are typically [Icon] or [Text] widgets. The selection state
```

Comment:
The boolean selection state of each widget is defined by the corresponding [isSelected] list item.